### PR TITLE
Fix IL bug created by replacing instructions 

### LINKED ISSUE #170: Rewriting bug resulting in invalid program
The following strange but simple program results into an "Common Language Runtime detected an invalid program" error after `coyote rewrite`. We should look into this and fix it (might be able to minimize more).

```csharp
[Test]
public static async Task TestRaces()
{
  Thread.Sleep(10);
  await Task.Run(async () =>
  {
    await Task.CompletedTask;
  });
}
```

### PR DESCRIPTION
**Pull request by lovettchris**
This fixes #170

This is one of those "how does anything work" bugs :-)

### CODE REVIEW

**Comment by pdeligia** [comment]
This is super neat and tidy @lovettchris! Yea makes sense that we should have been doing this check everywhere we mess up with the body and fix the instructions offsets, really nice catch!

File: Source/Test/Rewriting/Passes/AssemblyRewriter.cs [hunk 1]
```diff
@@ -632,5 +637,23 @@ protected static string GetFullyQualifiedMethodName(MethodReference method)
...
+        {
+            this.Processor.Replace(instruction, newInstruction);
+            foreach (var i in this.Processor.Body.Instructions)
+            {
+                if (i.Operand is Instruction j && j == instruction)
+                {
+                    i.Operand = newInstruction;
+                }
+            }
+        }
```
Comment (pdeligia) [inline Source/Test/Rewriting/Passes/AssemblyRewriter.cs:657]: Would it make sense to set `ModifiedMethodBody` to `true` here whenever we invoke this method, as we already do it explicitly?

**Comment by lovettchris** [comment]
Turns out this is an open issue in Mon.Cecil repo.  See https://github.com/jbevain/cecil/issues/182.

**Review by pdeligia — APPROVED** [review]

**Comment by pdeligia** [comment]
> Turns out this is an open issue in Mon.Cecil repo. See [jbevain/cecil#182](https://github.com/jbevain/cecil/issues/182).

Interesting nice find!

**Comment by pdeligia** [comment]
LGTM @lovettchris, thanks for this fix, let's merge :)

**Pull request merged**